public/app/features/dashboard/state/runRequest.ts

```diff
@@ -178,10 +210,29 @@ export function getProcessedDataFrames(results?: DataQueryResponseData[]): DataF
       }
     }
 
-    dataFrames.push(dataFrame);
+    switch (dataFrame.meta?.dataTopic) {
```

Comment:
I feel like the dashboard actually should not know about what topics exists. I think datasource should return data with different topics and then that should be used by the panel in what ever way it deems fit. This way you can define new topic if you create a custom panel and a custom data source.

Reply:
this is not part of the dashboard, but just request processing. The question is where should the response segregation (into topics) be done, here or inside the panels? (single dataFrame array to panels or split into different topic arrays). 

Not sure I like these topic arrays.

Reply:
> I feel like the dashboard actually should not know about what topics exists.

I meant it more in the sense that Grafana itself should not dictate the topics imho. Ie panel and dashboards as plugins themselves should be able to negotiate what topics they provide and what topics they could consume, Grafana only passing it through without needing to understand what those topics mean.